c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1150,7 +1150,7 @@ static std::string reportProcessMemoryInfo(int device) {
           pci_id, &nvml_device));
 
   std::vector<nvmlProcessInfo_v1_t> procs(8);
-  unsigned int size = procs.size();
+  auto size = procs.size();
```

Comment:
This implicit cache was necessary as removing the cast breaks the build it seems...

Reply:
Ok so back to `std::vector<decltype(nvmlProcessInfo_v1_t)>::size_type`?

Reply:
@SamuelMarks Did that actually not generate a compile error before? Does procs.size() not equal `std::vector<decltype(nvmlProcessInfo_v1_t)>::size_type`? It seems to on every platform except some versions of cuda? If so maybe make it clear with a static_cast / comment to explain why we are doing that. `auto size = static_cast<std::vector<decltype(nvmlProcessInfo_v1_t)>::size_type>(procs.size())` ?

Reply:
Hmm, well it looks like this variant worked on CI `std::vector<decltype(procs)>::size_type size`